wrangler-service/src/main/java/io/cdap/wrangler/service/directive/RecipeHandler.java

```diff
@@ -96,7 +114,41 @@ public void getRecipe(HttpServiceRequest request, HttpServiceResponder responder
                         @PathParam("context") String namespace,
                         @PathParam("id") String recipeId) {
     respond(responder, namespace, ns -> {
-      responder.sendString(GSON.toJson(recipeStore.getRecipe(new RecipeId(ns, recipeId))));
+      RecipeId id = RecipeId.builder(ns).setRecipeId(recipeId).build();
+      responder.sendString(GSON.toJson(recipeStore.getRecipe(id)));
+    });
+  }
+
+  @GET
+  @TransactionPolicy(value = TransactionControl.EXPLICIT)
+  @Path("v2/contexts/{context}/recipes")
+  public void listRecipes(HttpServiceRequest request, HttpServiceResponder responder,
+                          @PathParam("context") String namespace,
+                          @QueryParam("pageSize") Integer pageSize,
+                          @QueryParam("pageToken") String pageToken,
+                          @QueryParam("sortBy")String sortBy,
+                          @QueryParam("sortOrder") String sortOrder) {
+    if (sortBy != null && !(sortByMap.containsKey(sortBy))) {
+      throw new IllegalArgumentException("specified sortBy field is not supported");
```

Comment:
Can we also add the sortBy field name in the exception, so debugging would be easy.